frame/collective/src/lib.rs

```diff
@@ -168,12 +175,129 @@ decl_error! {
 		AlreadyInitialized,
 		/// The close call is made too early, before the end of the voting.
 		TooEarly,
+		/// There can only be a maximum of `MaxMembers`.
+		TooManyMembers,
+		/// There can only be a maximum of `MaxProposals` active proposals.
+		TooManyProposals,
+		/// The given weight bound for the proposal was too low.
+		WrongProposalBound,
+	}
+}
+
+/// Functions for calcuating the weight of dispatchables.
+mod weight_for {
+	use frame_support::weights::{RuntimeDbWeight, Weight};
+
+	/// Calculate the weight for `set_members`.
+	pub(crate) fn set_members(
```

Comment:
You can make these functions generic over `T: Trait`, then you don't need to pass `RuntimeDbWeight` each time.

Reply:
I think you code this way because you wanted to use `impl Into<Weight>`, and it seems not to work with explicit generic:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5d66f3d9abc97d1f1235f5bc70d50caa

maybe you can use regular generic type here instead of `impl Into<Weight>` 🤔

Reply:
Yeah, this just seemed the easiest way. I would need to have 1 generic argument per `impl Into<Weight>`. I'll try it out and see which I like better.

Reply:
Mmmh, making it generic over `T` leads to making the other params generic being fairly clumsy and harder to read. Not sure which tradeoff I like better here.
+ generic over `T`, other params concrete --> one value param less at call site, but more casting to `Weight`
+ generic over `T` and `Into<Weight>` --> very verbose `weight_for` function definitions that are hard to read IMO
+ pass in `db` and `impl Into<Weight>` --> one more param but easy generic other params

Reply:
I'm happy with this refactor https://github.com/paritytech/substrate/pull/5828/commits/bac6a08bf98dc43a63d4604d93972f3be83c1d92

Change all parameters to be of type `Weight` and call into when calling the function.